cppcache/src/TombstoneExpiryHandler.cpp

```diff
@@ -63,8 +63,7 @@ int TombstoneExpiryHandler::handle_timeout(const ACE_Time_Value& current_time,
           "[%s]",
           -sec / 1000 + 1, Utils::nullSafeToString(key).c_str());
       m_cacheImpl->getExpiryTaskManager().resetTask(
-          static_cast<long>(m_entryPtr->getExpiryTaskId()),
-          uint32_t(-sec / 1000 + 1));
+          m_entryPtr->getExpiryTaskId(), uint32_t(-sec / 1000 + 1));
```

Comment:
Why this type of cast?

Reply:
It got rid of the bad cast. Or are you asking about the second argument that is also casted?

Reply:
Yeah, the 2nd arg with the `type(expr)` cast.  Kinda seems like we should fix all the casts...

Reply:
There is a separate ticket for c-style casts.

The NOLINT will remain so that this method is ABI compatible with the ACE interface it is implementing.